L46-core-tls-credential-API.md

```diff
@@ -0,0 +1,390 @@
+  // debugging purposes. It should never be used in a production environment
+  // due to security concerns.
+  //
+  // @param tls_session_key_log_file_path: Path where tls session keys would
+  // be logged.
+  void set_tls_session_key_log_file_path(
+      const std::string& tls_session_key_log_file_path);
+  // Sets the certificate verifier used to perform post-handshake peer identity
+  // checks.
+  void set_certificate_verifier(
+      std::shared_ptr<CertificateVerifier> certificate_verifier);
+
+  // Sets the crl provider, see CrlProvider for more details.
+  void set_crl_provider(std::shared_ptr<CrlProvider> crl_provider);
+
+  // Sets the minimum TLS version that will be negotiated during the TLS
+  // handshake. If not set, the underlying SSL library will use TLS v1.2.
+  // @param tls_version: The minimum TLS version.
+  void set_min_tls_version(grpc_tls_version tls_version);
+  // Sets the maximum TLS version that will be negotiated during the TLS
+  // handshake. If not set, the underlying SSL library will use TLS v1.3.
+  // @param tls_version: The maximum TLS version.
+  void set_max_tls_version(grpc_tls_version tls_version);
+
+  // Sets a custom chain builder implementation that replaces the default chain
+  // building from the underlying SSL library.
+  void set_custom_chain_builder(std::shared_ptr<CustomChainBuilder>
+  chain_builder);
+}
+
+// Contains configurable options on the server side.
+class TlsServerCredentialsOptions final : public TlsCredentialsOptions {
+ public:
+  // Server side is required to use a provider, because server always needs to
+  // use identity certs.
+  explicit TlsServerCredentialsOptions(
+      std::shared_ptr<CertificateProviderInterface> certificate_provider)
+      : TlsCredentialsOptions() {
+    set_certificate_provider(certificate_provider);
+  }
+
+
+  // Sets option to request the certificates from the client.
+  // The default is GRPC_SSL_DONT_REQUEST_CLIENT_CERTIFICATE.
+  void set_cert_request_type(
+      grpc_ssl_client_certificate_request_type cert_request_type);
+}
+
+// Contains configurable options on the client side.
+// Client side doesn't need to always use certificate provider. When the
+// certificate provider is not set, we will use the root certificates stored
+// in the system default locations, and assume client won't provide any
+// identity certificates(single side TLS).
+class TlsChannelCredentialsOptions final : public TlsCredentialsOptions {
+ public:
+  // Sets the decision of whether to do a crypto check on the server certs.
+  // The default is true.
+  void set_verify_server_certs(bool verify_server_certs);
+
+// Builds TLS Credentials given TLS options.
+std::shared_ptr<ChannelCredentials> TlsCredentials(
+    const TlsChannelCredentialsOptions& options);
+}
+
+std::shared_ptr<ServerCredentials> TlsServerCredentials(
+    const grpc::experimental::TlsServerCredentialsOptions& options);
+```
+
+Here is an example of how to use this API:
```

Comment:
nit: "this API" is not well-defined since we've described a bunch of different APIs. Can we instead say something like "some of the APIs described above"?

Reply:
Changed to "Here is an example of how to configure channel credentials:"